app/js/lib/foxbox/foxbox.js

```diff
@@ -23,55 +19,33 @@ const p = Object.freeze({
   db: Symbol('db'),
   net: Symbol('net'),
   boxes: Symbol('boxes'),
-  isPollingEnabled: Symbol('isPollingEnabled'),
-  nextPollTimeout: Symbol('nextPollTimeout'),
   webPush: Symbol('webPush'),
-
-  // Private methods.
-  fetchServices: Symbol('fetchServices'),
-  getServiceInstance: Symbol('getServiceInstance'),
-  hasDoorLockChannel: Symbol('hasDoorLockChannel'),
+  api: Symbol('api'),
 });
 
-/**
- * Compare 2 objects. Returns true if all properties of object A have the same
- * value in object B. Extraneous properties in object B are ignored.
- * Properties order is not important.
- *
- * @param {Object} objectA
- * @param {Object} objectB
- * @return {boolean}
- */
-const isSimilar = (objectA, objectB) => {
-  for (let prop in objectA) {
-    if (!(prop in objectB) || objectA[prop] !== objectB[prop]) {
-      return false;
-    }
-  }
-
-  return true;
-};
-
 export default class Foxbox extends Service {
   constructor() {
     super();
 
     // Private properties.
     this[p.settings] = new Settings();
     this[p.db] = new Db();
-    this[p.net] = new Network(this[p.settings], (foxboxOnline) => {
-      this._dispatchEvent('box-online', foxboxOnline);
+
+    this[p.net] = new Network(this[p.settings]);
+    this[p.net].on('online-state-changed', (online) => {
+      this._dispatchEvent('online-state-changed', online);
     });
+
     this[p.boxes] = Object.freeze([]);
-    this[p.isPollingEnabled] = false;
-    this[p.nextPollTimeout] = null;
-    this[p.webPush] = new WebPush(this[p.net], this[p.settings],
-      (msg) => {
-        this._dispatchEvent('push-action', msg);
-      });
+    this[p.api] = new API(this[p.net], this[p.settings]);
 
-    // Public properties.
-    this.recipes = null;
+    this[p.webPush] = new WebPush(this[p.api], this[p.settings]);
+    this[p.webPush].on('message-received', (msg) => {
+      this._dispatchEvent('push-message-received', msg);
+    });
```

Comment:
Maybe move the events listeners to the `init()` method instead of the constructor.
This may provide some optimisation where we delay the costly init after a basic view is rendered.
We could also add a `teardown` method where we remove these event listeners.